www/src/pages/table/forms/TransformationForm.tsx

```diff
@@ -0,0 +1,304 @@
+import { ActionType } from "@ant-design/pro-table";
+import { useRef, useState } from "react";
+import Editor from "@monaco-editor/react";
+import { TransformationTable, useStores } from "../../../models";
+import {
+  TPayload
+} from "../../../models/transformation/TransformationStore";
+
+const styles = {
+};
+
+export interface TransformationFormProps {
+  type: string;
+  tableId: number;
+}
+
+export const TransformationForm = withStyles(styles)(
+  observer(
+    ({
+      classes,
+      tableId,
+    }: TransformationFormProps & WithStyles<typeof styles>) => {
+      const onDone = () => Modal.destroyAll();
+      const actionRef = useRef<ActionType>();
+      const [form] = Form.useForm();
+      const { transformationStore } = useStores();
+
+      const options = {
+        autoIndent: "full",
+        contextmenu: true,
+        fontFamily: "monospace",
+        fontSize: 13,
+        lineHeight: 24,
+        matchBrackets: "always",
+        minimap: {
+          enabled: false,
+        },
+        scrollbar: {
+          horizontalSliderSize: 4,
+          verticalSliderSize: 18,
+        },
+        selectOnLineNumbers: true,
+        roundedSelection: false,
+        readOnly: false,
+        cursorStyle: "line",
+        automaticLayout: true,
+      };
+      const [result, setResult] = useState<TransformationTable[] | undefined>();
+      const [editorData, setEditorData] = useState<string | undefined>();
+
+      const columns = [
+        { dataIndex: "row_id", title: "Row ID" },
+        { dataIndex: "previous_value", title: "Previous Value" },
+        {
+          dataIndex: "result",
+          title: "Result",
+          render: (transformed_value: string | boolean) => {
+            if (
+              typeof transformed_value === "string" &&
+              transformed_value.includes("Traceback")
+            ) {
+              return (
+                <>
+                  {" "}
+                  {
+                    <Tag color={"volcano"}>
+                      <pre>{transformed_value}</pre>
+                    </Tag>
+                  }
+                </>
+              );
+            } else {
+              return transformed_value;
+            }
+          },
+        },
+      ];
+
+      const onExecute = async () => {
+        const transformPayload = new TPayload();
+        transformPayload.type = form.getFieldValue("type");
+        transformPayload.code = editorData;
+        transformPayload.mode = "restrictedpython";
+        transformPayload.datapath = form.getFieldValue("datapath");
+        transformPayload.outputpath = form.getFieldValue("outputpath");
+        transformPayload.tolerance = form.getFieldValue("tolerance");
+        transformPayload.rows = form.getFieldValue("rows");
+
+        let response = await transformationStore.postData(tableId, transformPayload);
+        setResult(response);
+      };
+
+      function handleEditorChange(value: string | undefined) {
```

Comment:
antdesign has a way to do this automatically with form. checkout the example code in here https://ant.design/components/form#formitemusestatus, and migrate the editorData to that.

Reply:
resolved it